pennylane/measurements/mid_measure.py

```diff
@@ -218,6 +232,18 @@ def _merge(self, other: "MeasurementValue"):
         merged_measurement_ids = list(set(self.measurement_ids).union(set(other.measurement_ids)))
         merged_measurement_ids.sort()
 
+        merged_measurements = []
+        for item in self.measurements + other.measurements:
+            is_duplicate = False
+            for unique_item in merged_measurements:
+                if item is unique_item:
+                    is_duplicate = True
+
+            if not is_duplicate:
+                merged_measurements.append(item)
+
+        merged_measurements.sort(key=lambda m: m.id)
```

Comment:
I guess re: my previous comment, it is useful here? though I'm not sure how sorting matters other than consistency seeing as the IDs are random in ordering

Reply:
The sorting is needed for correctly indexing the arguments to the merged processing function. But, a list of ids is not needed for that.

Reply:
that works for me!